codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -287,7 +287,7 @@ public ChannelFuture writePushPromise(ChannelHandlerContext ctx, int streamId, i
     @Override
     public ChannelFuture writeGoAway(ChannelHandlerContext ctx, int lastStreamId, long errorCode, ByteBuf debugData,
             ChannelPromise promise) {
-        return lifecycleManager.writeGoAway(ctx, lastStreamId, errorCode, debugData, promise);
+        return lifecycleManager.closeConnectionRemote(ctx, lastStreamId, errorCode, debugData, promise);
```

Comment:
Actually, I guess depending on how you look at it this terminology could be correct, since you're indicating that the remote side can no longer create new streams. Maybe `close` isn't the right term here as it's too easily confused with the logic of `closeStreamRemote` method.  It also might be confused with closing the channel, which is different as well.

Maybe it would be better to stick with something like `ChanelFuture goAway(...)`, rather than trying to disguise the GOAWAY logic behind another method name.  WDYT?